src/API/processAPIData.js

```diff
@@ -3,76 +3,96 @@ import peoplePathName from "../peoplePathName";
+		rating: changeVoteFormat(movie.vote_average),
+		vote_count: movie.vote_count,
+		namedGenres: nameGenres(movie.genre_ids, genres),
+		poster: `${posterURL}${movie.poster_path}`,
+	}));
 };
 
 export const processMovieDetailsData = (movieDetails) => {
-    return (
-        {
-            id: movieDetails.id,
-            title: movieDetails.title,
-            releaseDate: movieDetails.release_date,
-            releaseYear: changeDateFormat(movieDetails.release_date),
-            rating: changeVoteFormat(movieDetails.vote_average),
-            voteCount: movieDetails.vote_count,
-            description: movieDetails.overview,
-            production: getCountryName(movieDetails.production_countries),
-            genres: getGenreName(movieDetails.genres),
-            backdrop: `${backdropURL}${movieDetails.backdrop_path}`,
-            poster: `${posterURL}${movieDetails.poster_path}`,
-        }
-    )
+	return {
+		id: movieDetails.id,
+		title: movieDetails.title,
+		releaseDate: movieDetails.release_date,
+		releaseYear: changeDateFormat(movieDetails.release_date),
+		rating: changeVoteFormat(movieDetails.vote_average),
+		voteCount: movieDetails.vote_count,
+		description: movieDetails.overview,
+		production: getCountryName(movieDetails.production_countries),
+		genres: getGenreName(movieDetails.genres),
+		backdrop: `${backdropURL}${movieDetails.backdrop_path}`,
+		poster: `${posterURL}${movieDetails.poster_path}`,
+	};
 };
 
 export const processSearchResults = (rawResults, rawGenreList, path) => {
+	switch (path) {
+		case moviesPathName:
+			return processMovieListData(rawResults, rawGenreList);
+		case peoplePathName:
+			return rawResults.results;
+		default:
+			return {};
+	}
+};
 
-    switch (path) {
-        case moviesPathName:
-            return processMovieListData(rawResults, rawGenreList);
-        case peoplePathName:
-            return rawResults.results;
-        default:
-            return {};
-    };
-};
\ No newline at end of file
+export const processPeopleCastMoviesData = (castMovies, genreList) => {
+	const genres = genreList.genres;
+
+	return castMovies.map((movie) => ({
+		id: movie.id,
+		title: movie.title,
+		date: changeDateFormat(movie.release_date),
+		rating: changeVoteFormat(movie.vote_average),
+		vote_count: movie.vote_count,
+		release_date: movie.release_date,
+		namedGenres: nameGenres(movie.genre_ids, genres),
+		poster: movie.poster_path ? `${posterURL}${movie.poster_path}` : null,
+	}));
+};
+
+export const processPeopleCrewMoviesData = (crewMovies, genreList) => {
+	const genres = genreList.genres;
+
+	return crewMovies.map((movie) => ({
+		id: movie.id,
+		title: movie.title,
+		date: changeDateFormat(movie.release_date),
+		rating: changeVoteFormat(movie.vote_average),
+		vote_count: movie.vote_count,
+		release_date: movie.release_date,
+		namedGenres: nameGenres(movie.genre_ids, genres),
+		poster: movie.poster_path ? `${posterURL}${movie.poster_path}` : null,
+	}));
+};
```

Comment:
I think the code is more organized now, so it's a good job that you created these functions